python/cudf/cudf/utils/dtypes.py

```diff
@@ -144,16 +144,13 @@ def numeric_normalize_types(*args):
 
 
 def is_numerical_dtype(obj):
-    if is_categorical_dtype(obj):
+    # TODO: we should handle objects with a `.dtype` attribute,
+    # e.g., arrays, here.
+    try:
+        dtype = np.dtype(obj)
```

Comment:
What if someone gives us a Pandas nullable integer type?

Reply:
We certainly aren't handling this currently. On branch-0.19

```python
>>> cudf.Series([1, 2, 3], dtype=pd.Int64Dtype())  # TypeError
>>> cudf.utils.dtypes.is_numerical_dtype(pd.Int64Dtype()) # TypeError
```

I agree we _shouldl_ support this. But how to do so in an efficient way is a difficult question. @vyasr and I were talking about this a couple of days ago, and he has some ideas for how to make dtype introspection faster/cheaper. We can perhaps take on this problem there?